portal/server/helpers/pageSpecificValidationErrors.js

```diff
@@ -4,126 +4,115 @@
-  - this is currently only used for Companies House validation.
+ ALWAYS_SHOW_ERROR_FIELDS are fields that are:
+ - not required (therefore optional)
+ - BUT if this field has an error, always show it after form submit.
+ - this is the opposite of 'page specific validation rules'
+ - this is currently only used for Companies House validation.
 */
 const hasSubmittedAlwaysShowErrorFields = (allFields, submittedFields) => {
-  const { ALWAYS_SHOW_ERROR_FIELDS } = allFields;
+ const { ALWAYS_SHOW_ERROR_FIELDS } = allFields;
 
   const hasAlwaysShowFields = ALWAYS_SHOW_ERROR_FIELDS && ALWAYS_SHOW_ERROR_FIELDS.length > 0;
 
-  if (hasAlwaysShowFields) {
+ if (hasAlwaysShowFields) {
     const pageFields = Object.keys(submittedFields).filter((fieldName) =>
       ALWAYS_SHOW_ERROR_FIELDS.includes(fieldName));
 
-    if (pageFields.length > 0) {
-      return true;
-    }
-  }
+   if (pageFields.length > 0) {
+     return true;
+   }
+ }
 
-  return false;
+ return false;
 };
 
 const mapAlwaysShowErrorFields = (validationErrors, fields) => {
-  const mappedErrors = validationErrors;
+ const mappedErrors = { ...validationErrors };
 
-  mappedErrors.errorList = filterErrorList(validationErrors.errorList, fields.ALWAYS_SHOW_ERROR_FIELDS);
+ const filteredErrorList =
+filterErrorList(validationErrors.errorList,
+fields.ALWAYS_SHOW_ERROR_FIELDS);
 
-  return {
-    ...generateErrorSummary(
-      mappedErrors,
-      errorHref,
-    ),
-  };
+ return {
+   ...generateErrorSummary({ ...mappedErrors, errorList:
+filteredErrorList }, errorHref),
+ };
 };
 
 const mapRequiredAndAlwaysShowErrorFields = (validationErrors, allFields) => {
-  const mappedErrors = validationErrors;
-  const allRequiredFields = requiredFieldsArray(allFields);
-  const alwaysShowErrorFields = allFields.ALWAYS_SHOW_ERROR_FIELDS;
-
-  const fieldsToReturn = [
-    ...allRequiredFields,
-    ...alwaysShowErrorFields,
-  ];
-
-  mappedErrors.errorList = filterErrorList(validationErrors.errorList, fieldsToReturn);
-
-  return {
-    ...generateErrorSummary(
-      mappedErrors,
-      errorHref,
-    ),
-  };
-};
+ const mappedErrors = cloneDeep(validationErrors);
+ const allRequiredFields = requiredFieldsArray(allFields);
+ const alwaysShowErrorFields =
+allFields.ALWAYS_SHOW_ERROR_FIELDS?allFields.ALWAYS_SHOW_ERROR_FIELDS:[];
+
+ const fieldsToReturn = [...allRequiredFields, ...alwaysShowErrorFields];
+
+ mappedErrors.errorList = filterErrorList(validationErrors.errorList,
+fieldsToReturn);
 
+ return {
+   ...generateErrorSummary(mappedErrors, errorHref),
+ };
+};
 const pageSpecificValidationErrors = (validationErrors, fields, submittedFields) => {
   if (validationErrors && validationErrors.errorList) {
     if (!submittedFields.viewedPreviewPage && hasSubmittedAlwaysShowErrorFields(fields, submittedFields)) {
+      if(submittedFields.status==='Incomplete'){
+        return mapRequiredAndAlwaysShowErrorFields(validationErrors, fields);
```

Comment:
Out of interest why does this fix things please?

Reply:
If supplier page is not completed means status in incomplete. It will check the status and display the error(required field and optional field error)